### ISSUE TITLE
Libraries dependency troubles in 2.0.1.18 - stop updating

### ISSUE THREAD

**Issue by mnaiman** [duplicati/duplicati#1891]
Hi, I think it is necessary to release 2.0.1.19. 

NuGet building introduced some dependency troubles which have similar results in issues: #1883 #1890 #1887 

This rendering update 2.0.1.18 unusable in many scenarios.
@kenkendk Please either remove 2.0.1.18 from update server or this needs to be fixed fast, before more people will be affected.

Unfortunate is that 2.0.1.18 upgraded DB, so it is not possible to downgrade. :(

**Comment by mr-flibble** [comment]
Same problem https://github.com/duplicati/duplicati/issues/1883 - please remove 2.0.1.1.8, because after update is not possible backup/restore

**Comment by mnaiman** [comment]
AFAIK
1) AlphaFS.dll is wrong version 1.5 (should be 2.0)
2) alphavss folder is missing (library should be updated but this is another story)
3) Duplicati.Library.Backend.Mega.dll is dependent on MegaApiClient.dll which is dependent on Newtonsoft.Json.dll version 6, but version 9 is latest and installed
4) SharpAESCrypt.exe in dependent on .NET 2.0 not 4.5 (no trouble now, but on servers 2.0 is not even default installed - maybe cause of #1887)

Maybe more projects in solution are affected. This is result of my short analysis.

**Comment by RightOnTrack** [comment]
I was experiencing a constant loop of verifying the files. Yes, version 2.0.1.18 needs to be pulled. I did downgrade successfully however. Here is what to do in Windows.

1. Export .json file so you can import later.

2. Make sure "Hidden items" is checked off under "View" in Windows 8-10, or show hidden files in "Folder options" in Win 7.

3. Uninstall Duplicati 2.
 
4. Go to C:\ProgramData and delete the Duplicati 2 folder.

5. Go to C:\Users\"Your User Account"\AppData\Local and delete Duplicati 2 folder. Repeat steps 4-5 for any other User Accounts in Windows.

6. Reboot and install Duplicati 2.0.1.1.7

7. Import the .json file you saved earlier.

8. Recreate and Repair the local database.

Now just go ahead and run a backup, it should complete it.

**Comment by NickPyz** [comment]
@RightOnTrack  I am reviewing the procedure you posted so I can version downgrade to 2.0.1.17. Can you please clarify the file name and path of the *.json file you refer to in Step 1? 

I assumed that individual backup job configurations are contained inside the various .sqlite files in \AppData\Roaming\Duplicati\. Has 2.0.1.18 changed these files in a way that makes them unusable in 2.0.1.17?

Thx.

**Comment by agrajaghh** [comment]
@NickPyz If you click on the BackupJob and chose "Export" and then "File" you can download a json file with the job configuration.

You're right, the config is saved in the DB, but if the DB format changes during an update, a downgrade isnt possible anymore. (I don't know if thats the case with the update from 17 to 18)

**Comment by RightOnTrack** [comment]
@NickPyz When you save the .json file, the filename is the name of the backup job and it will save to the default download location on your pc, usually either download or desktop folder.

@agrajaghh This is the error I received after downgrading to 2.0.1.17. on 8-14. However, a recreate and repair fixed it. Still working fine as of today.

System.Exception: Found 103 remote files that are not recorded in local storage, please run repair
at Duplicati.Library.Main.Operation.FilelistProcessor.VerifyRemoteList(BackendManager backend, Options options, LocalDatabase database, IBackendWriter log)
at Duplicati.Library.Main.Operation.BackupHandler.PreBackupVerify(BackendManager backend)
at Duplicati.Library.Main.Operation.BackupHandler.Run(String[] sources, IFilter filter)
at Duplicati.Library.Main.Controller.<Backup>c__AnonStorey0.<>m__0(BackupResults result)
at Duplicati.Library.Main.Controller.RunAction[T](T result, String[]& paths, IFilter& filter, Action`1 method)   at Duplicati.Library.Main.Controller.Backup(String[] inputsources, IFilter filter)
at Duplicati.Server.Runner.Run(IRunnerData data, Boolean fromQueue)

**Comment by NickPyz** [comment]
@RightOnTrack  @agrajaghh 

Your instructions were effective. I was able to downgrade to 2.0.1.17. And I recovered my numerous backup jobs using the export / import .json files procedure -  followed by "Recreate (delete and repair)" of each DB. So far, each job works error free.

Thanks.

**Comment by vb0** [comment]
For me it looks like the Duplicati folder was in Appdata\Roaming and called just Duplicati (without 2).
Installed from msi, .17 was also the first version I installed (I'm new to duplicati). Windows 10 x64.
There was just "Recreate" possible (as there was no database available, I don't know if I removed the wrong folder or something else was at play) anyway it seems to be working as expected.

**Comment by kenkendk** [comment]
@mnaiman: Thanks for the details!

It turns out it was a problem with my machine having the old version of AlphaFS.
Even though a newer version was requested, it would use the old version.

I have changed the build script to remove all the local folders before making a build.

For alphavss, the dll files are correctly added to the snapshot library, but not copied into the other projects that depend on this library. I fixed the build script to copy these files in manually.

I have released a new 2.0.1.19 build right now that has these fixes.
